File: z-clients/javascript/src/apis/adminMetrics.ts
// this file is @generated

import {
    type GetMetricsOut,
    GetMetricsOutSerializer,
} from '../models/getMetricsOut';
import { HttpMethod, DiomRequest, type DiomRequestContext } from "../request";

export class AdminMetrics {
    public constructor(private readonly requestCtx: DiomRequestContext) {}

    /** Dump the current metrics (which would otherwise be sent to the OTLP metrics receiver) */
    public get(
    ): Promise<GetMetricsOut> {
        const request = new DiomRequest(HttpMethod.GET, "/api/v1.admin.metrics.get");

        
        return request.send(
            this.requestCtx,
            GetMetricsOutSerializer._fromJsonObject,
        );
    }
}

File: z-clients/rust/src/api/admin_metrics.rs
// this file is @generated
use crate::{Configuration, error::Result, models::*};

pub struct AdminMetrics<'a> {
    cfg: &'a Configuration,
}

impl<'a> AdminMetrics<'a> {
    pub(super) fn new(cfg: &'a Configuration) -> Self {
        Self { cfg }
    }

    /// Dump the current metrics (which would otherwise be sent to the OTLP metrics receiver)
    pub async fn get(&self) -> Result<GetMetricsOut> {
        crate::request::Request::new(http::Method::GET, "/api/v1.admin.metrics.get")
            .execute(self.cfg)
            .await
    }
}
